scripts/lib/CIME/XML/test_reporter.py

```diff
@@ -0,0 +1,120 @@
+"""
+Interface to the testreporter xml.  This class inherits from GenericXML.py
+
+"""
+
+from CIME.XML.standard_module_setup import *
+from CIME.XML.generic_xml import GenericXML
+from CIME.utils import expect,get_model
+
+import urllib
+
+
+
+class TestReporter(GenericXML):
+
+    def __init__(self):
+        """
+        initialize an object
+        """
+
+        expect(get_model() == 'cesm', "testreport is only meant to populate the CESM test database." )
+        self.root = None
+
+        GenericXML.__init__(self)
+
+    def setup_header(self, tagname,machine,compiler,mpilib,testroot,testtype,baseline):
+        #
+        # Create the XML header that the testdb is expecting to recieve 
+        #
+        tlelem    = ET.Element("testrecord")
+        elem      = ET.Element('tag_name')
+        elem.text = tagname
+        tlelem.append(elem)
+        elem                = ET.Element('mach')
+        elem.text           = machine
+        tlelem.append(elem)
+        elem                = ET.Element('compiler',attrib={"version":""})
+        elem.text           = compiler
+        tlelem.append(elem)
+        elem                = ET.Element('mpilib',attrib={"version":""})
+        elem.text           = mpilib
+        tlelem.append(elem)
+        elem                = ET.Element('testroot')
+        elem.text           = testroot
+        tlelem.append(elem)
+        elem                = ET.Element('testtype')
+        elem.text           = testtype
+        tlelem.append(elem)
+        elem   = ET.Element('baselinetag')
+        elem.text   = baseline
+        tlelem.append(elem)
+    
+        self.root=tlelem
+
+
+
+    def dumpxml(self):
```

Comment:
why have an additional function here?  Just call GenericXML.write directly or if you really want the default outfile name at least call the inherited function write instead of dumpxml.